apis/projectcontour/v1/httpproxy.go

```diff
@@ -152,13 +152,15 @@ type Route struct {
 	//
 	// +kubebuilder:validation:Optional
 	PathRewrite *PathRewritePolicy `json:"pathRewritePolicy,omitempty"`
+	// The policy for managing headers during proxying
+	// +optional
+	RequestHeaderPolicy *RequestHeaderPolicy `json:"requestHeaderPolicy,omitempty"`
```

Comment:
Should the type here simply be `HeaderPolicy` or `HeaderRewritePolicy`?  What about the contents is request-specific?  I could see the validation being different, but that's done separately, and could still be made context sensitive in other ways (where it matters).